oscore-cf/src/main/java/org/eclipse/californium/oscore/HashMapCtxDB.java

```diff
@@ -209,6 +213,19 @@ private static String normalizeServerUri(String uri) throws OSException {
 			LOGGER.error("Error in the request URI: " + uri + " message: " + e.getMessage());
 			throw new OSException(e.getMessage());
 		}
+		
+		//Further normalization for IPv6 addresses
```

Comment:
I'm not sure, if this would be effected by

https://github.com/eclipse/californium/blob/2.0.x/californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java#L613

as well.

Reply:
Yes, it may be. I will check this.

Reply:
I have now added a workaround for this bug. Previously this bug could be triggered if the uri contained an unsupported character in the scope. I now solve it in a similar way to the link you provided. The unsupported characters in the scope are removed before getting the host portion of the address. I then also added code so the original scope is restored because it is needed for the continued parsing. This was done in commit d8a66fc2510b9a6c3081a8e13f114d931c9672d0